general/array.hpp

```diff
@@ -17,6 +17,7 @@
 #include "device.hpp"
 #include "error.hpp"
 #include "globals.hpp"
+#include "forall.hpp" // for MFEM_FORALL_SWITCH
```

Comment:
So far, we've kept `forall.hpp` out of `mfem.hpp` -- do we really want to include it now?

We can move `forall.hpp` in the `.cpp` file -- keep the inline implementation for `use_dev = false` and call a method defined in the `.cpp` file for the case `use_dev = true`.

Reply:
`done` in [a09dbef](https://github.com/mfem/mfem/pull/2568/commits/a09dbef03eca79330ec1a173fab9386e74e1c20e)

Reply:
This leads to the need of specific `template class Array<...>;` instanciations: trivials `bool`, `unsigned long long`, and non-trivial ones `mfem::KnotVector*`, `mfem::IntegrationRule*`, etc.
Not sure what's best.

Reply:
Hmm, yes, I didn't think about this issue. Maybe we can implement this for any type (since we assume the type is trivial) by passing the host address and bytes-size of the `value` and the destination pointer?